src/index.ts

```diff
@@ -190,7 +208,7 @@ export function parse(str: string, options: ParseOptions = {}): Token[] {
         prefix,
         suffix: "",
         pattern: pattern || defaultPattern,
-        modifier: tryConsume("MODIFIER") || ""
+        modifier: tryConsumeModifier() || "",
```

Comment:
@wanderview Are you intentionally allowing an asterisk to be followed by another modifier? URLPattern doesn't support that AFAIK...